File: strategy_pattern/java/1-duck-behavior/MiniDuckSimulator.java


public class MiniDuckSimulator {

    public static void main(String[] args) {
        Duck mallard = new MallardDuck();
        mallard.display();
        mallard.fly();
        mallard.quack();
        mallard.setFlyBehavior(new FlyRocketPowered());
        mallard.fly()

        Duck decoy = new DecoyDuck();
        decoy.display();
        decoy.fly();
        decoy.quack();
        decoy.setQuackBehavior(new Quack());
        decoy.quack();
    }
}

File: strategy_pattern/python/1-duck-behavior/mini_duck_simulator.py
from duck import MallardDuck, DecoyDuck


def main():
    mallard = MallardDuck()
    mallard.display()
    mallard.quack()
    mallard.fly()

    decoy = DecoyDuck()
    decoy.display()
    decoy.quack()
    decoy.fly()


if __name__ == '__main__':
    main()